src/DGmethods/DGmodel.jl

```diff
@@ -175,176 +157,149 @@ Initialize the ODE state array.
+  N    = polynomialorder(grid)
+  Np   = dofs_per_element(grid)
+  vgeo = grid.vgeo
   topology = grid.topology
+  Ω    = topology.realelems
+  nΩ   = length(Ω)
+
   # FIXME: Remove after updating CUDA
-  h_vgeo = Array(grid.vgeo)
+  h_vgeo = Array(vgeo)
   DFloat = eltype(h_vgeo)
-  Np = dofs_per_element(grid)
-  DA = arraytype(grid)
+  DA     = arraytype(grid)
 
   weights = view(h_vgeo, :, grid.Mid, :)
   weights = reshape(weights, size(weights, 1), 1, size(weights, 2))
 
-  state = MPIStateArray{Tuple{Np, num_state(bl,DFloat)}, DFloat, DA}(topology.mpicomm,
-                                               length(topology.elems),
-                                               realelems=topology.realelems,
-                                               ghostelems=topology.ghostelems,
-                                               vmaprecv=grid.vmaprecv,
-                                               vmapsend=grid.vmapsend,
-                                               nabrtorank=topology.nabrtorank,
-                                               nabrtovmaprecv=grid.nabrtovmaprecv,
-                                               nabrtovmapsend=grid.nabrtovmapsend,
-                                               weights=weights,
-                                               commtag=commtag)
-
-  auxstate = param.aux
-  dim = dimensionality(grid)
-  polyorder = polynomialorder(grid)
-  vgeo = grid.vgeo
-  device = typeof(state.Q) <: Array ? CPU() : CUDA()
-  nrealelem = length(topology.realelems)
-  @launch(device, threads=(Np,), blocks=nrealelem,
-          initstate!(bl, Val(dim), Val(polyorder), state.Q, auxstate.Q, vgeo, topology.realelems, args...))
-  MPIStateArrays.start_ghost_exchange!(state)
-  MPIStateArrays.finish_ghost_exchange!(state)
-
-  return state
+  Y = MPIStateArray{Tuple{Np, num_state(bl,DFloat)}, DFloat, DA}(topology.mpicomm,
+      length(topology.elems),
+      realelems=topology.realelems,
+      ghostelems=topology.ghostelems,
+      vmaprecv=grid.vmaprecv,
+      vmapsend=grid.vmapsend,
+      nabrtorank=topology.nabrtorank,
+      nabrtovmaprecv=grid.nabrtovmaprecv,
+      nabrtovmapsend=grid.nabrtovmapsend,
+      weights=weights,
+      commtag=commtag)
+
+  device = typeof(Y.Q) <: Array ? CPU() : CUDA()
+
+  @launch(device, threads=(Np,), blocks=nΩ,
+          initstate!(bl, Val(Nᵈ), Val(N), Y.Q, α.Q, vgeo, Ω, args...))
+
+  MPIStateArrays.start_ghost_exchange!(Y)
+  MPIStateArrays.finish_ghost_exchange!(Y)
+
+  return Y
 end
 
 
 """
     dof_iteration!(dof_fun!::Function, R::MPIStateArray, disc::DGBalanceLaw,
-                   Q::MPIStateArray)
+                   Y::MPIStateArray)
 
 Iterate over each dof to fill `R` using the `dof_fun!`. The syntax of the
 `dof_fun!` is
 ```
-dof_fun!(l_R, l_Q, l_Qvisc, l_aux)
+dof_fun!(l_R, l_Y, l_σ, l_aux)
 ```
-where `l_R`, `l_Q`, `l_Qvisc`, and `l_aux` are of type `MArray` filled initially
+where `l_R`, `l_Y`, `l_σ`, and `l_α` are of type `MArray` filled initially
```

Comment:
Same as above: `l_Ydiff` (or similar) and `l_aux` is better. May even consider `state` in variable name.

Reply:
Isn't the alpha subscript what is used in the design docs?